### PR TITLE
chore(test): Add Kamelet's custom from test cases

### PR DESCRIPTION
**Pull request by lordrip**
### Context
This PR adds a couple of missing test cases for when the source definitions are `empty` or `undefined`. It also adds a bit of type safety.

### CODE REVIEW

File: packages/ui/src/utils/get-custom-schema-from-actual-kamelet-schema.ts [hunk 1]
```diff
@@ -1,50 +1,58 @@
...
-    type: kamelet.metadata.labels['camel.apache.org/kamelet.type'],
-    icon: kamelet.metadata.annotations['camel.apache.org/kamelet.icon'],
-    supportLevel: kamelet.metadata.annotations['camel.apache.org/kamelet.support.level'],
-    catalogVersion: kamelet.metadata.annotations['camel.apache.org/catalog.version'],
-    provider: kamelet.metadata.annotations['camel.apache.org/provider'],
-    group: kamelet.metadata.annotations['camel.apache.org/kamelet.group'],
-    namespace: kamelet.metadata.annotations['camel.apache.org/kamelet.namespace'],
-  };
+export const getCustomSchemaFromActualKameletSchema = (kamelet: IKameletDefinition): IKameletCustomDefinition => {
+  const name = getValue(kamelet, 'metadata.name', '');
```
Comment (shivamG640) [inline packages/ui/src/utils/get-custom-schema-from-actual-kamelet-schema.ts:12]: For better readability, write  `getValue(kamelet, 'metadata.name') ` instead!

**Review by shivamG640 — APPROVED** [review]
